### ISSUE TITLE
redelivery has failed when try to use ip address for queue name

### ISSUE THREAD

**Issue by HyunHaechang** [guidesmiths/rascal#166]
I tried to use ip address for queue name, and met this error. 

In advanced example, I just changed save_user's queue name from "registration_service:user:save" to  "agent_10.10.10.10"
and try to redelivery from nack, some error is occured.

TypeError: Unknown type to encode: undefined
    at encodeFieldValue (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:229:15)
    at encodeArray (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:97:19)
    at encodeFieldValue (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:162:23)
    at encodeTable (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:85:21)
    at encodeFieldValue (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:171:23)
    at encodeTable (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:85:21)
    at encodeFieldValue (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:171:23)
    at encodeTable (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\codec.js:85:21)
    at encodeBasicProperties (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\defs.js:2755:11)
    at module.exports.encodeProperties (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\defs.js:3526:12)
    at Connection.C.sendMessage (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\connection.js:544:16)
    at ConfirmChannel.C.sendMessage (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\channel.js:91:26)
    at ConfirmChannel.Channel.publish (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\callback_model.js:171:17)
    at ConfirmChannel.publish (E:\workspace\nodejs\robot-agent\node_modules\amqplib\lib\callback_model.js:301:36)
    at Object.callback (E:\workspace\nodejs\robot-agent\node_modules\rascal\lib\amqp\SubscriberError.js:91:28)
    at E:\workspace\nodejs\robot-agent\node_modules\rascal\node_modules\async\dist\async.js:1529:26

**Comment by cressie176** [comment]
Hi @HyunHaechang,

That's a good find. The issue is [here](https://github.com/guidesmiths/rascal/blob/master/lib/amqp/SubscriberError.js#L75) where I use lodash.set to store the recovery information in a message header. Because the queue name includes a combination of period and numerical characters (i.e. 10.10.10.10), lodash is creating an array.

I'll see if I can find an alternative which doesn't do this, but in the meantime maybe you can workaround by using a different format for the queue name, e.g. agent_10_10_10_10

**Comment by cressie176** [comment]
Looks like if I change my use of _.set to the array notation it might work as desired

**Comment by HyunHaechang** [comment]
I really appreciate your quick advice help and your great works.

**Comment by cressie176** [comment]
Posting for comedy value - this is document that lodash.set is creating. There are 3 nested arrays (one per instance of `.10`), each with 10 elements, the first 9 of which are `null`

```json
{
  "contentType": "text/plain",
  "headers": {
    "rascal": {
      "originalQueue": "622e3d95-c690-4dee-be08-9d6800592c8d:q_10.10.10.10",
      "originalVhost": "/",
      "redeliveries": 0,
      "recovery": {
        "622e3d95-c690-4dee-be08-9d6800592c8d:q_10": [
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          [
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              {
                "republished": 1
              }
            ]
          ]
        ]
      }
    }
  },
  "deliveryMode": 1,
  "messageId": "2577303c-38ec-4c24-8dfa-e55db3387bd9"
}
```

**Comment by cressie176** [comment]
Fix on the way

**Comment by cressie176** [comment]
Fix published in rascal@13.1.2
